subprojects/gdk_rust/gdk_electrum/src/lib.rs

```diff
@@ -989,33 +877,47 @@ impl<S: Read + Write> Headers<S> {
         Ok(())
     }
 }
-impl<S: Read + Write> Syncer<S> {
-    pub fn sync(&mut self) -> Result<bool, Error> {
+
+#[derive(Default)]
+struct DownloadTxResult {
+    txs: Vec<(Txid, BETransaction)>,
+    unblinds: Vec<(elements::OutPoint, Unblinded)>,
+}
+
+impl Syncer {
+    pub fn sync(&self, client: &Client) -> Result<bool, Error> {
         trace!("start sync");
         let start = Instant::now();
 
         let mut history_txs_id = HashSet::new();
         let mut heights_set = HashSet::new();
         let mut txid_height = HashMap::new();
+        let mut scripts = HashMap::new();
 
-        let mut last_used = [0u32; 2];
-        let mut wallet_chains = vec![0,1];
+        let mut last_used = Indexes::default();
+        let mut wallet_chains = vec![0, 1];
         wallet_chains.shuffle(&mut thread_rng());
-        for i in wallet_chains {
-            let int_or_ext = Index::from(i)?;
+        for i in 0..=1 {
```

Comment:
This isn't actually using the shuffled `wallet_chains`, looks like this got lost when rebasing.

Reply:
good spot c45e4799575cd18f639a7853fc1a7440ae6ae42a